### PR TITLE
Student to rfid assocation

### PR DESCRIPTION
**Pull request by gearedmonkey**
### Summary

You can now scan an rfid card and associate it with an existing student.

### Checklist

- [x] Have you added unit and functional tests as appropriate?
- [x] Did you update/add documentation for new methods or changed functionality?
- [x] Have you merged the target branch down into this one?

### How to Review

1. Start client on this branch and server on student-exists-tff207
2. In the omnibar, enter: 123456
3. Verify that John shows up(he is already associated with an rfid)
4. In the omnibar, enter: 147852
5. Verify input modal is up.
6, Enter studentID: 159753(meet Mr Rufflekins!)
7. Click associate button
8. Confirm associate toast appears
9. In the omnibar enter: 147852
10. Verify Mr. Rufflekins shows up.
11. Go to view all students page and confirm Mr. Rufflekins has an rfid associated.

### Links

- [TFF-207](https://msoese.atlassian.net/browse/TFF-207)
- TheFourFifths/consus/pull/102

### CODE REVIEW

File: test/unit/controllers/components/omnibar.js [hunk 1]
```diff
@@ -67,21 +67,6 @@ describe("OmnibarController", () => {
 
         });
 
-        it('Dispatches "ERROR" when student is not found',()=>{
```
Comment (gearedmonkey) [inline test/unit/controllers/components/omnibar.js]: This test no longer makes sense. On a student not found the rfid workflow will trigger.

File: src/controllers/components/omnibar.js [hunk 2]
```diff
@@ -22,55 +22,53 @@ export default class OmnibarController {
         hashHistory.push(route);
     }
 
-    static getStudent(id) {
-        if (typeof id === 'string') {
-            id = parseInt(id);
+    static getStudent(rfid) {
+        if (typeof rfid === 'string') {
+            rfid = parseInt(rfid);
```
Comment (mattmahn) [inline src/controllers/components/omnibar.js]: See https://github.com/TheFourFifths/consus/pull/102#discussion_r109319286

File: src/views/components/omnibar.jsx [hunk 3]
```diff
@@ -2,31 +2,41 @@ import React from 'react';
...
         };
     }
 
     changeQuery(e) {
         let regex = new RegExp("^[a-zA-Z0-9]*$");
         let regexOnlyNums = new RegExp("^[0-9]*$");
-        if(regex.test(e.target.value)) {
-            if (e.target.value.length === 6 && regexOnlyNums.test(e.target.value)) {
+        let rfid = e.target.value;
+        if(regex.test(rfid)) {
```
Comment (mattmahn) [inline src/views/components/omnibar.jsx]: See https://github.com/TheFourFifths/consus/pull/102#discussion_r109319286

File: test/unit/controllers/pages/student.js [hunk 4]
```diff
@@ -292,6 +300,43 @@ describe("StudentController", () => {
...
+                })
+            );
+            StudentController.studentToRfid(123456, 123456).then(() => {
+                assert.isTrue(createRfidToStudentAssosciation.called);
+                assert.isTrue(dispatcherSpy.called);
+                assert.strictEqual(dispatcherSpy.getCall(0).args[0], "ERROR");
+            })
+        });
+
+        it('Dispatches "ERROR" on failed association ', () => {
```
Comment (mattmahn) [inline test/unit/controllers/pages/student.js]: NACK: These names are backwards

File: test/functional/view-models.js [hunk 5]
```diff
@@ -4,7 +4,7 @@ import { assert } from 'chai';
 import MockServer from '../util/mock-server';
 import models from '../test-cases/models';
 
-describe('View all models', function () {
+describe.only('View all models', function () {
```
Comment (tondef001) [inline test/functional/view-models.js]: And this is why I should really read the test files more carefully...

Comment (gearedmonkey) [inline test/functional/view-models.js]: maybe this is why travis fails..

Comment (tondef001) [inline test/functional/view-models.js]: I removed it locally and an entirely new test is breaking

Comment (gearedmonkey) [inline test/functional/view-models.js]: That doesn't sound ideal.

**Pull request merged**